### PR TITLE
Turn on the logging span exporter if agent debug is enabled.

### PR DESCRIPTION
**Pull request by jkwatson**

### CODE REVIEW

File: javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentInitializer.java [hunk 1]
```diff
@@ -47,6 +47,7 @@
   // fields must be managed under class lock
   private static ClassLoader AGENT_CLASSLOADER = null;
 
+  // called via reflection in the OpenTelemetryAgent class
```
Comment (jkwatson) [inline javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentInitializer.java:50]: I was working on understanding the startup process, and noticed that IDEA said this was unused, so it seemed worthwhile to add a comment. Can pull out of this PR if not appropriate.

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/Config.java [hunk 2]
```diff
@@ -159,4 +159,8 @@ public Properties asJavaProperties() {
     properties.putAll(getAllProperties());
     return properties;
   }
+
+  public boolean isAgentDebugEnabled() {
+    return getBooleanProperty("otel.javaagent.debug", false);
```
Comment (breedx-splk) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/Config.java:164]: Can we get a test to cover this?  Might seem silly, but it helps to confirm the key name and also the default value.

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java [hunk 3]
```diff
@@ -72,7 +86,7 @@ private static synchronized void installExportersFromJar(
         getExporterFactory(SpanExporterFactory.class, exporterLoader);
 
     if (spanExporterFactory != null) {
-      installExporter(spanExporterFactory, config, builder);
+      installSpanExporter(spanExporterFactory, config, builder);
```
Comment (breedx-splk) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java:89]: 👍

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java [hunk 4]
```diff
@@ -44,6 +46,18 @@ public void configure(SdkTracerProviderBuilder sdkTracerProviderBuilder) {
...
 
     maybeConfigureExporterJar(sdkTracerProviderBuilder);
+    maybeEnableLoggingExporter(sdkTracerProviderBuilder);
+  }
+
+  private static void maybeEnableLoggingExporter(SdkTracerProviderBuilder builder) {
+    if (Config.get().isAgentDebugEnabled()) {
+      // todo: is this really a reliable way to detect this?
+      // don't install another instance if the user has already explicitly requested it.
+      if (!Config.get().getProperty("otel.traces.exporter", "").equalsIgnoreCase("logging")
```
Comment (breedx-splk) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java:56]: This is hard to read.  First thought was that putting the `Config.get()` into a local var might help...but also, what about factoring out a method called `hasLoggingExporter()` or something similar?

**Review by breedx-splk — CHANGES_REQUESTED** [review]
Cool stuff...I want this!
A couple small asks from the review.

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/Config.java (hunk 2 above)
Comment (jkwatson) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/Config.java:164]: done

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java (hunk 4 above)
Comment (jkwatson) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java]: done. check out the refactoring

Comment (breedx-splk) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java]: Whoopsie....it now reads "if logging is enabled, enable logging".  I think the method might want to be called `loggingExporterIsNotAlreadyConfigured` if you don't wanna invert the logic.

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java [hunk 5]
```diff
@@ -44,6 +46,23 @@ public void configure(SdkTracerProviderBuilder sdkTracerProviderBuilder) {
...
+  private static void maybeEnableLoggingExporter(SdkTracerProviderBuilder builder) {
+    if (Config.get().isAgentDebugEnabled()) {
+      // don't install another instance if the user has already explicitly requested it.
+      if (loggingExporterIsAlreadyConfigured()) {
+        builder.addSpanProcessor(SimpleSpanProcessor.create(new LoggingSpanExporter()));
+      }
+    }
+  }
+
+  private static boolean loggingExporterIsAlreadyConfigured() {
```
Comment (jkwatson) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java:61]: oh yeah, this should be "loggingExporterIsNotAlreadyConfigured()"

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java [hunk 6]
```diff
@@ -44,6 +46,23 @@ public void configure(SdkTracerProviderBuilder sdkTracerProviderBuilder) {
...
+    // todo: is this really a reliable way to detect this?
+    Config config = Config.get();
+    return !config.getProperty("otel.traces.exporter", "").equalsIgnoreCase("logging")
+        && !config.getProperty("otel.exporter", "").equalsIgnoreCase("logging");
```
Comment (anuraaga) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java:65]: I don't think we support this property anywhere anymore.

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java [hunk 7]
```diff
@@ -44,6 +46,23 @@ public void configure(SdkTracerProviderBuilder sdkTracerProviderBuilder) {
...
+    if (Config.get().isAgentDebugEnabled()) {
+      // don't install another instance if the user has already explicitly requested it.
+      if (loggingExporterIsAlreadyConfigured()) {
+        builder.addSpanProcessor(SimpleSpanProcessor.create(new LoggingSpanExporter()));
+      }
+    }
+  }
+
+  private static boolean loggingExporterIsAlreadyConfigured() {
+    // todo: is this really a reliable way to detect this?
```
Comment (anuraaga) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java:62]: Seems ok

**Review by anuraaga — APPROVED** [review]
Thanks!

**Review by iNikem — APPROVED** [review]
Method should still be renamed

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java (hunk 4 above)
Comment (jkwatson) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java]: fixed

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java (hunk 5 above)
Comment (jkwatson) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java]: fixed

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java (hunk 6 above)
Comment (jkwatson) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java]: removed

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java (hunk 7 above)
Comment (jkwatson) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentTracerProviderConfigurer.java]: todo removed

**Review by breedx-splk — APPROVED** [review]

**Review by trask — APPROVED** [review]
Thanks!

**Pull request merged**